src/core/config/openthread-core-config-check.h

```diff
@@ -34,6 +34,8 @@
 #ifndef OPENTHREAD_CORE_CONFIG_CHECK_H_
 #define OPENTHREAD_CORE_CONFIG_CHECK_H_
 
+#include "config/device_type_check.h"
```

Comment:
Is it still necessary after we have included "config/device_type_check.h" at the beginning of `openthread-core-config.h`